### DISCUSSION TITLE
Complete all Server Commands

### DISCUSSION THREAD

**Discussion by angelhdzmultimedia** [nuxtrdev/nuxtr-vscode#124] [Ideas]
Same as with https://github.com/nuxtrdev/nuxtr-vscode/discussions/123,
I'm willing to contribute with Server Plugin Command (Nitro) .

I noticed if I right click on `~/server/plugins` it allows to add a plugin, but only a client plugin in `~/plugins` instead of `~/server/plugins`.
![image](https://github.com/nuxtrdev/nuxtr-vscode/assets/10191085/833826b3-06db-49c8-86bf-730aa3d2a15f)

**Comment by angelhdzmultimedia** [comment]
### Update
Not to saturate the Discussion, as this is also related to server, I will be willing to work on the rest of the Server Commands.

Right-clicking `~/server` dir should have `Add Server Api` (`~/server/api`), `Add Server Route` (`~/server/routes`), `Add Server Middleware` (`~/server/middleware`) and `Add Server Plugin` (`~/server/plugins`).

**Comment by angelhdzmultimedia** [comment]
### Update
Just noticed all the server commands are done already https://github.com/nuxtrdev/nuxtr-vscode/blob/main/src/commands/Nitro.ts
but they don't appear on VSCode Command Palette, or by right-clicking `~/server`... 🤔

**Comment by adhamfarrag** [comment]
Hi @angelhdzmultimedia,

Thanks for your all great ideas :) 

Right clicking on `~/server` was one option, but giving you 5 options `New API/Route/Plugin/etc..` wouldn't look nice on your context menu and requires more attention to choose what you want. So we made context menus appear on each respective directory. After creating Nitro [directories](https://nitro.unjs.io/guide)  manually, context menu items should appear. 

Also, all commands are supported and can be found using `Nuxtr: Nitro` prefix.

![2024-03-14 at 2  51 31@2x](https://github.com/nuxtrdev/nuxtr-vscode/assets/23562275/f63c6bbb-5351-47b7-964e-7ec65b6aec67)

Tell me if `Nitro` was misleading to you.

**Reply by angelhdzmultimedia** [reply]
Yeah then it's a bug, I don't get those Nitro prefixes commands in VSCode command palette, and right-clicking server corresponding dirs create assets for client only and I get an error notification. I'll investigate, thank you.

**Reply by adhamfarrag** [reply]
@angelhdzmultimedia If they do not show up it means Nuxtr isn't activated for you. Happy to help with that.

**Reply by angelhdzmultimedia** [reply]
I have Nuxtr v0.2.15.

- I open a sales.code-workspace in VSCode with the following config:

```json
{
  "folders": [
    {
      "name": "client",
      "path": "client"
    },
    {
      "name": "server",
      "path": "server"
    }
  ],

  "settings": {
    "nuxtr.monorepoMode.DirectoryName": "${workspaceFolder}\\client",
    "typescript.tsdk": "node_modules\\typescript\\lib"
  }
}
```
- It effectively opens as a workspace, with `client` and `server` multi-root directories.

- I right-click `client/server/api`
- Click Add API Endpoint in the context menu.
- API endpoint is generated successfully but with error:
![image](https://github.com/nuxtrdev/nuxtr-vscode/assets/10191085/1ba06a79-00a5-4b6a-ae89-3853377ef42d)

- I right-click `client/server/routes`
- Add Endpoint or Add Server Route or Add Route does not appear in the context menu.
- I right-click `client/server/middleware`
-  Click Add Middleware in the context menu.
- A prompt asks for default or global (and it shouldn't because it's a Server Middleware).
- A client middleware is created in `client/middleware` instead and get the error:
![image](https://github.com/nuxtrdev/nuxtr-vscode/assets/10191085/0ffe0e20-ea8d-4113-837f-1ee210bfed00)

- I open VSCode command palette
- I type "Nitro"
- No Nitro related command in the results:
![image](https://github.com/nuxtrdev/nuxtr-vscode/assets/10191085/4f5c3d49-ef04-4311-a896-087bb0b2367b)

**Reply by angelhdzmultimedia** [reply]
I open `client` dir in VScode, and then I get the Nitro commands:
![image](https://github.com/nuxtrdev/nuxtr-vscode/assets/10191085/2c959a7d-587a-4f38-8f05-3122e58f8432)

Also all server commands are working:
![image](https://github.com/nuxtrdev/nuxtr-vscode/assets/10191085/38015345-de24-4c61-a59c-96cfa81e966e)


So the issue is Nuxtr extension even though is provided with the `"nuxtr.monorepoMode.DirectoryName": "${workspaceFolder}\\client"` or `"nuxtr.monorepoMode.DirectoryName": "client"` setting, it's not working correctly for Multi-root workspaces apparently.

**Note:**
servers/routes still not shows Add Server Route or Add Nitro Route in the context menu, but `Nuxtr: Create new Nitro Route` appears in command palette.

**Edit** 
I had a `Nuxt (unofficial)` extension installed which caused some confusions. Just uninstalled.

**Reply by angelhdzmultimedia** [reply]
Ok I'm back at the beginning. 

No context menu command or command palette command shows up in Multi-root workspace mode.

**Reply by angelhdzmultimedia** [reply]
Apparently the extension activates only if it finds a `nuxt.config.ts`, so the `nuxtr.monorepoMode.DirectoryName` setting is not doing anything.

**Reply by angelhdzmultimedia** [reply]
Well, had to open 2 workspaces or else I won't finish my app. Hope Nuxtr works for Multi-root Workspace in the future. Thanks.
